src/app/context/SettingsContext.tsx

```diff
@@ -24,17 +25,22 @@ export const SettingsProvider = ({
 }: {
   children: React.ReactNode;
 }) => {
-  const [settings, setSettings] =
-    useState<SettingsContextType["settings"]>(DEFAULT_SETTINGS);
+  const [settings, setSettings] = useState<
+    SettingsContextType["settings"] | undefined
+  >();
   const [isLoading, setIsLoading] = useState(true);
   const [currencyRate, setCurrencyRate] = useState(0);
 
   // call this to trigger a re-render on all occassions
   const updateSetting = async (setting: Setting) => {
-    setSettings((prevState) => ({
-      ...prevState,
-      ...setting,
-    }));
+    if (!setting) return;
+    setSettings((prevState) => {
+      if (prevState)
```

Comment:
if the default is `{}` (same type of the actual value) then this would not be needed?